_source/_posts/2022-03-14-react-vite-number-converter.md

```diff
@@ -0,0 +1,350 @@
+
+```bash
+npm run dev
+```
+
+If you are used to React applications built with `create-react-app`, you might expect to wait a few seconds before the development server starts. With Vite, I see the following message in less than a second.
+
+```
+  vite v2.8.4 dev server running at:
+
+  > Local: http://localhost:3000/
+  > Network: use `--host` to expose
+
+  ready in 461ms.
+```
+
+You can now open your browser at `http://localhost:3000/` and test the application. When I click the **Converter** link in the navigation bar, I see something like this.
+
+{% img blog/react-vite-number-converter/react-vite-application.jpg alt:"The number conversion utility" width:"512" %}{: .center-image }
+
+I opened up the Developer Tools in Chrome to understand how Vite achieves these fast starting-up times. When you open up the **Network** tab and filter by JS requests, you will see many requests to individual JavaScript sources. You will even see the `.tsx` sources you just edited. 
+
+{% img blog/react-vite-number-converter/react-vite-modules.jpg alt:"Vite development server under the hood." width:"800" %}{: .center-image }
+
+I then looked closer at `App.tsx,` and saw what is shown in the above image. When the development server is asked to serve a `.tsx` file, it will compile it on the fly into browser-compatible code. But it keeps all the `import` statements in place and uses the support for ES modules in the newer browsers. This way, the bundling stage is eliminated, and the loading times are significantly reduced.
+
+## Adding authentication with Okta to the application
+
+A secure application needs user authentication to keep unauthorized users out of restricted areas. With Okta, it is easy to add authentication to your Vite application in just a few steps. 
+
+{% include setup/cli.md type="spa" framework="React" loginRedirectUri="http://localhost:3000/callback" logoutRedirectUri="http://localhost:3000" %}
+
+Now you are ready to add the Okta libraries for React into your project. Open the terminal in the project folder and run the following command.
+
+```bash
+npm install -E @okta/okta-react@6.4.2 @okta/okta-auth-js@6.1.0
+```
+
+Vite needs some help resolving import aliases used by the Okta libraries. To make things work, open `vite.config.ts` and add the following entry to the configuration object.
+
+```ts
+resolve: {
+  alias: [
+    {
+      find: "@okta/okta-auth-js",
+      replacement: require.resolve("@okta/okta-auth-js/dist/okta-auth-js.umd.js"),
+    },
+  ],
+}
+```
+
+If your IDE or build can't resolve `require`, you may need to add the `@types/node` library for the `require` method. In the terminal, add the library by running the following command:
+
+```shell
+npm i –save-dev @types/node
+```
+
+Now, open `src/main.tsx` and add the following code immediately after the `import` statements.
+
+```ts
+import { Security } from "@okta/okta-react";
+import { OktaAuth, toRelativeUrl } from '@okta/okta-auth-js';
```

Comment:
Imports should use consistent quotes. For example, double or single quotes. I prefer single quotes with JavaScript myself.